platooning_strategic_IHP/src/platoon_manager_ihp.cpp

```diff
@@ -343,11 +345,11 @@ namespace platoon_strategic_ihp
         // If the distance headway between the subject vehicle and its predecessor is an issue
         // according to the "min_gap" and "max_gap" thresholds, then it should follow its predecessor
         // The following line will not throw exception because the length of downtrack array is larger than two in this case
-        double distHeadwayWithPredecessor = downtrackDistance[downtrackDistance.size() - 2] - downtrackDistance[downtrackDistance.size() - 1];
+        double distHeadwayWithPredecessor = downtrackDistance[hostPosInPlatoon_ - 2] - downtrackDistance[hostPosInPlatoon_ - 1];
```

Comment:
I think this needs to be 
downtrackDistance[hostPosInPlatoon_ - 1] - downtrackDistance[hostPosInPlatoon]
If host is 2nd car in platoon then hostPostInPlatoon_ - 2 is -1, which will blow up the vector.